supervise.go

```diff
@@ -554,12 +550,49 @@ func killSupervisedServices() {
 		s.state.Set(Stopping)
 
 		s.setStopped(true)
-		s.Signal(syscall.SIGTERM)
+		s.Signal(signal)
+		states = append(states, s.state)
 	}
+	return states
+}
 
-	for _, s := range services.S {
-		s.waitTillStopped()
+// killSupervisedServices is called before rebooting when upgrading, allowing
+// processes to terminate in an orderly fashion.
+func killSupervisedServices(signalDelay time.Duration) {
+	termStates := signalSupervisedServices(syscall.SIGTERM)
```

Comment:
Can you add a log message before sending SIGTERM, too, please?

Reply:
Added `sending sigterm to all services`